<?xml version="1.0"?>
<launch>

    <arg name="dev_front" default="/dev/ttyACM1"/>
    <arg name="dev_back" default="/dev/ttyACM1"/>
    <arg name="baud" default="38400"/>
    <arg name="address_front" default="128"/>
    <arg name="address_back" default="129"/>
    <arg name="max_speed" default="1.7"/>
    <arg name="ticks_per_meter" default="89478.2"/>
    <arg name="run_diag" default="true"/>

    <node if="$(arg run_diag)" pkg="roboclaw_node" type="complete_node_b.py" name="roboclaw_node_b">
        <param name="~dev_front" value="$(arg dev_front)"/>
        <param name="~dev_back" value="$(arg dev_back)"/>
        <param name="~baud" value="$(arg baud)"/>
        <param name="~address_front" value="$(arg address_front)"/>
        <param name="~address_back" value="$(arg address_back)"/>
        <param name="~max_speed" value="$(arg max_speed)"/>
        <param name="~ticks_per_meter" value="$(arg ticks_per_meter)"/>
    </node>

    <node pkg="diagnostic_aggregator" type="aggregator_node"
          name="diagnostic_aggregator_b">
        <rosparam command="load"
                  file="$(find roboclaw_node)/config/roboclaw_diag.yaml"/>
    </node>

</launch>



